abi/codecForCustomTypes_enum.go

```diff
@@ -0,0 +1,61 @@
+package abi
+
+import (
+	"bytes"
+	"fmt"
+	"io"
+)
+
+func (c *codec) encodeNestedEnum(writer io.Writer, value EnumValue) error {
+	err := c.doEncodeNested(writer, U8Value{Value: value.Discriminant})
+	if err != nil {
+		return err
+	}
+
+	for _, field := range value.Fields {
+		err := c.doEncodeNested(writer, field.Value)
+		if err != nil {
+			return fmt.Errorf("cannot encode field '%s' of enum, because of: %w", field.Name, err)
+		}
+	}
+
+	return nil
+}
+
```

Comment:
Took me a lot of time to review and analyse the encoding. This is very good! :muscle: